numba/ir.py

```diff
@@ -767,7 +836,7 @@ def list_vars(self):
         return [self.contextmanager]
 
 
-class Arg(AbstractRHS):
+class Arg(AbstractRHS, EqualityCheckMixin):
```

Comment:
mixin class `EqualityCheckMixin ` should go before the base class `AbstractRHS` so that mro will look at the mixin before the base class.